tests/experimental/datasources/test_postgres_datasource.py

```diff
@@ -37,17 +42,24 @@ def _source(
     validate_batch_spec: Callable[[SqlAlchemyDatasourceBatchSpec], None],
     dialect: str,
     connection_string: str = "postgresql+psycopg2://postgres:@localhost/test_ci",
+    engine: Optional[sqlalchemy.engine.Engine] = None,
 ) -> PostgresDatasource:
     execution_eng_cls = sqlachemy_execution_engine_mock_cls(
         validate_batch_spec=validate_batch_spec, dialect=dialect
     )
     original_override = PostgresDatasource.execution_engine_override
     try:
         PostgresDatasource.execution_engine_override = execution_eng_cls
-        yield PostgresDatasource(
-            name="my_datasource",
-            connection_string=connection_string,
+        with mock.patch("sqlalchemy.create_engine"):
+            postgres_datasource = PostgresDatasource(
+                name="my_datasource",
+                connection_string=connection_string,
+                engine=engine,
+            )
+        postgres_datasource.engine = MockSaEngine(
```

Comment:
We could do something similar to this with `execution_engine_override`. Then we won't be able to make this immutable though.